packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -809,6 +871,56 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
         );
     }
 
+    /**
+     * @notice Overrides BALANCE.
+     * @param _contract Address of the contract to query the OVM_ETH balance of.
+     * @return _BALANCE OVM_ETH balance of the requested contract.
+     */
+    function ovmBALANCE(
+        address _contract
+    )
+        override
+        public
+        returns (
+            uint256 _BALANCE
+        )
+    {
+        // Easiest way to get the balance is query OVM_ETH as normal.
+        bytes memory balanceOfCalldata = abi.encodeWithSelector(
+            IUniswapV2ERC20.balanceOf.selector,
+            _contract
+        );
+
+        // Static call because this should be a read-only query.
+        (bool success, bytes memory returndata) = ovmSTATICCALL(
+            gasleft(),
+            OVM_ETH_ADDRESS,
+            balanceOfCalldata
+        );
+
+        // All balanceOf queries should successfully return a uint, otherwise this must be an OOG.
```

Comment:
Have you confirmed that this is true for a balance of 0? I would assume so, but I think worth double-checking in remix.